src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,60 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private Random random = new Random();
+    private Figure figure;
+    private ColorSupplier colorSupplier = new ColorSupplier();
+
+    public Figure getDefaultFigure() {
+        Figures[] values = Figures.values();
+        switch (values[random.nextInt(values.length)].toString()) {
+            case "CIRCLE":
+                figure = new Circle();
+                break;
+            case "ISOSCELES_TRAPEZOID":
+                figure = new IsoscelesTrapezoid();
+                break;
+            case "RECTANGLE":
+                figure = new Rectangle();
+                break;
+            case "RIGHT_TRIANGLE":
+                figure = new RightTriangle();
+                break;
+            case "SQUARE":
+                figure = new Square();
+                break;
+            default:
+        }
+        figure.setColor(colorSupplier.getRandomColor());
+        return figure;
+    }
+
+    public Figure getRandomFigure() {
+        Figures[] values = Figures.values();
+        float random1 = 20 * random.nextFloat();
+        float random2 = 20 * random.nextFloat();
+        float random3 = 20 * random.nextFloat();
+        switch (values[random.nextInt(values.length)].toString()) {
+            case "CIRCLE":
+                figure = new Circle(random1);
+                break;
+            case "ISOSCELES_TRAPEZOID":
+                figure = new IsoscelesTrapezoid(random1, random2, random3);
+                break;
+            case "RECTANGLE":
+                figure = new Rectangle(random1, random2);
+                break;
+            case "RIGHT_TRIANGLE":
+                figure = new RightTriangle(random1, random2);
+                break;
+            case "SQUARE":
+                figure = new Square(random1);
+                break;
+            default:
+        }
+        figure.setColor(colorSupplier.getRandomColor());
+        return figure;
```

Comment:
let's create a switch case that will depend on a random number from 0 to 4. For example, in the case of 0, you can return a random circle and so on ...